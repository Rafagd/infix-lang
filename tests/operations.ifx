add is {
    left + right
};

void println 1 add 2;

mul is {
    (index  is i32) = 0;
    (result is i32) = 0;
    (index < right) repeat {
        result = result + left;
    };
    result
};

early_return is {
    (right > 0) ? {
        void println "EARLY";
        void return void;
    };
    void println "LATE";
};

void early_return -1;
void early_return 1;

