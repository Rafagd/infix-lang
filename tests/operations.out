3
6
LATE
EARLY
